test/parallel/test-net-server-close-before-dns.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+const common = require('../common');
+const net = require('net');
+
+net.createServer(() => {}).listen(common.PORT, 'localhost', () => {
+  net.connect(common.PORT, () => {
+    throw new Error('a connection was made');
+  });
```

Comment:
Should also be able to check `this.address()` and see if it returns the connection information.